lib/ansible/module_utils/eos.py

```diff
@@ -139,12 +141,16 @@ def connect(self, **kwargs):
 
         username = self.module.params['username']
         password = self.module.params['password']
+        key_filename = self.module.params['ssh_keyfile']
 
         try:
             self.shell = Shell(CLI_PROMPTS_RE, CLI_ERRORS_RE)
-            self.shell.open(host, port=port, username=username, password=password)
+            if key_filename:
+                self.shell.open(host, port=port, username=username, key_filename=key_filename)
+            else:
+                self.shell.open(host, port=port, username=username, password=password)
```

Comment:
Thank you, I thought something still looked suspicious here. Really, there's no need for any check at all, paramiko uses key_filename first, and fails to password anyway.